thorlcr/thorutil/thormisc.cpp

```diff
@@ -1371,8 +1371,12 @@ class CRowServer : public CSimpleInterface, implements IThreaded, implements IRo
 #ifdef TRACE_GLOBAL_GROUP
         ActPrintLog(activity, "%s", __func__);
 #endif
-        running = false;
-        comm.cancel(RANK_ALL, mpTag);
+        bool wanted = true;
+        if (running.compare_exchange_strong(wanted, false))
+        {
+            running = false;
```

Comment:
trivial: not needed (it will be set to false by the compare_exchange

Reply:
removed.